WeakAuras/GenericTrigger.lua

```diff
@@ -824,15 +900,32 @@ function GenericTrigger.LoadDisplays(toLoad, loadEvent, ...)
     genericTriggerRegisteredEvents[event] = true;
   end
 
+  for unit, events in pairs(unitEventsToRegister) do
+    for event in pairs(events) do
+      if not frame.units[unit] then
+        frame.units[unit] = CreateFrame("FRAME")
+        frame.units[unit]:SetScript("OnEvent", HandleUnitEvent);
+      end
+      xpcall(frame.units[unit].RegisterUnitEvent, trueFunction, frame.units[unit], event, unit)
+      genericTriggerRegisteredUnitEvents[unit] = genericTriggerRegisteredUnitEvents[unit] or {};
+      genericTriggerRegisteredUnitEvents[unit][event] = true;
+    end
+  end
+
   for id in pairs(toLoad) do
     GenericTrigger.ScanWithFakeEvent(id);
   end
 
   if (eventsToRegister[loadEvent]) then
     WeakAuras.ScanEvents(loadEvent, ...);
   end
+  local loadUnit = ...
+  if loadUnit and unitEventsToRegister[loadUnit] and unitEventsToRegister[loadUnit][loadEvent] then
+    WeakAuras.ScanUnitEvents(loadEvent, ...);
```

Comment:
UNIT_FLAGS, UNIT_ENTERED_VEHICLE, UNIT_EXITED_VEHICLE are used for loading auras